src/common/http.cc

```diff
@@ -681,6 +681,11 @@ serveFile(ResponseWriter& response, const char* fileName, const Mime::MediaType&
         throw HttpError(Code::Internal_Server_Error, "");
     }
 
+    res = ::close(fd);
```

Comment:
I think that the fd should also be closed if the `fstat` call above fails.

Actually,  while thinking about it, maybe could we benefit from a scope guard kind of construct to make sure the fd is correctly closed on all possible paths when an exception arises.

We would have something like `Scope_Failure(::close(fd));` which will close the fd in its destructor if unwinding occurs (see [std::uncaught_exception](http://en.cppreference.com/w/cpp/error/uncaught_exception))